chaindataloader/sideload.go

```diff
@@ -0,0 +1,73 @@
+package chaindataloader
+
+import (
+	"fmt"
+
+	"github.com/btcsuite/btcd/wire"
+)
+
+const (
+
+	// ErrEndOfFile is returned when we have reached the end of a file while reading it.
+	ErrEndOfFile = sideLoadErr("end of file reached")
+)
+
+// sideLoadErr is a type that represents the defined errors in the chaindataloader package.
+type sideLoadErr string
+
+// Error returns the error in string form.
+func (e sideLoadErr) Error() string {
+	return string(e)
+}
+
+// Reader is the interface a caller fetches data from.
+type Reader[messageType any | *wire.BlockHeader] interface {
+
+	// NextHeader fetches header from a chaindataloader source.
+	NextHeader() (messageType, error)
+
+	// EndHeight indicates the height of the last Header in the chaindataloader source.
+	EndHeight() int64
+
+	// StartHeight indicates the height of the first Header in the chaindataloader source.
+	StartHeight() int64
+
+	// SetHeight sets the required offset to fetch the next header after the header
+	// at "height".
+	SetHeight(height int64) error
+
+	// HeadersChain returns the bitcoin network the headers in the chaindataloader source belong
+	// to.
+	HeadersChain() wire.BitcoinNet
+}
+
+// ReaderConfig is a struct that contains configuration details required to fetch a Reader.
+type ReaderConfig struct {
+
+	// SourceType is the format type the chaindataloader source is.
+	SourceType string
```

Comment:
Is this meant to be read from the command line? If so, it's missing struct tags.

Reply:
https://github.com/lightninglabs/neutrino/pull/285#discussion_r1375169882